### ISSUE TITLE
[FEATURE] [PLUGIN] Add Chromecast Support

### ISSUE THREAD

**Issue by troycarpenter** [icefields/Power-Ampache-2#144]
It would be nice if the app could cast music to Chromecast enabled devices.

At home, it is not uncommon for us to cast music and podcasts from our mobile player for everyone to enjoy across our Chromecast speakers.

**Comment by icefields** [comment]
Chromecast support is on my roadmap, but it's not the top priority at the moment. I'm developing a separate installable plugin for Chromecast and Android Auto.

The challenge is that the libraries for these platforms are proprietary, and I want to keep PA2 fully open-source and free of trackers and ads. Any Google-related feature would require Play Services, which involves Google trackers and will compromise your privacy. To address this, these features will be available through an optional plugin that I’m developing, installable separately.

The FDroid version of Power Ampache 2 will never have those feature, since FDroid doesn't allow the use of closed source libraries.

**Comment by troycarpenter** [comment]
> Totally understand. Having a plugin downloadable separately for free is the cleaner and most scalable (ie. updated to google features won't affect the main app) way to develop Google-related features without affecting the open-source and track-free features, while at the same time making everyone happy (hopefully). Work has started already, but I'm currently prioritizing the widget, tablet UI and multi-user, plus the usual bug fixes.
> 
> Thanks for spreading the word about Power Ampache 2. Use this promo code to download Power Ampache 2 from the play store for free from your work phone (If you want, you can always make a donation later): AHUKQKHMM655NTX6X03JDAF
> 
> ps. It's single-use. Use it before anyone else sees this!

Thank you very much.  Downloaded just fine and now I can test on my daily use phone.

**Comment by potagerGit** [comment]
+1 for Android Auto Feature Request !

**Comment by icefields** [comment]
I predict that I will start testing the plugin in about a month

**Comment by Maxi2555** [comment]
Hello, Is the Chromecast Plugin already avaliable, and do I need the Playstore version to use it?

**Comment by icefields** [comment]
@Maxi2555 I'm still working on it, I think I will release a beta around the end of January. 
You don't need the Playstore app, any version of the app will work. The plugin itself won't be available on FDdroid because it contains proprietary libraries, but it will be free to download from the PlayStore and Github.

**Comment by icefields** [comment]
@Maxi2555 not yet, I keep pushing it back because of other priorities. It's proceeding slowly, but we're almost there.

**Comment by icefields** [comment]
@Maxi2555 @troycarpenter I’ve just released the first beta version of the Chromecast plugin in the Telegram group chat.
For now, beta versions will only be available through the group and the Matrix Space. If you're not part of it yet, join here https://t.me/PowerAmpache2 ,to download the APK and try it out on your device.

Otherwise, you’ll need to wait for the official release. This plugin won’t be uploaded to F-Droid, as it includes proprietary Google libraries. However, it will be available through all the other usual channels: Play Store, Telegram, Matrix Space, and GitHub Releases.

**Comment by icefields** [comment]
**Chromecast Plugin Update**

The updated Chromecast plugin has been submitted to the Play Store, but due to a new Google policy, I need at least 12 users to test it via early access before it can be officially released. Unfortunately, there's no way around this requirement.

Since the plugin relies on Google's proprietary Chromecast library, it cannot be published on F-Droid.

So far, I've had 6 testers sign up (huge thanks to those who have). But until we reach the 12-user minimum, the release will remain on hold. If you're interested in testing or know someone who might be, your help would be greatly appreciated. Otherwise, this could take a few more weeks.

**Comment by Maxi2555** [comment]
How do I join the Beta? Can't find it where it usually is

**Comment by icefields** [comment]
hi @Maxi2555 , if you go 2 comments up (comment from Sept 2), that post contains all the info you need to get the beta version of the plugin. In short, the beta versions is only available on Telegram. The official release, whenever I'm able to unblock this situation, available everywhere.

Anyway, my latest comment is not about beta-testing, it's about the new mandatory google mandate which dictates that all apps, including PA2 plugins, need at least 12 early access tester to be approved for release. 
I'm at 7 testers today, if you'd like to help I just need the email you use with the PlayStore so I can add you to the list. 
You can send it to me by email, or contact me in the Telegram group chat and I'll DM you.
Thanks!
